File: Chapter 02/C#/2.3.3 Testability/RealDatabaseConnection.cs
using System;
using System.Collections.Generic;

namespace Chapter02.Testability;

public class RealDatabaseConnection : IDatabaseConnection
{
    private readonly string _connectionString;

    public RealDatabaseConnection(string connectionString)
    {
        _connectionString = connectionString;
        Console.WriteLine($"\n  [DB] Connecting to... {_connectionString}");
    }

    public List<string> GetData(string query)
    {
        return new List<string> { "real_data_row1", "real_data_row2" };
    }
}

File: Chapter 02/Java/src/main/java/com/GrokkingSoftwareArchitecture/Chapter 02/2.3.3 Testability/RealDatabaseConnection.java
package com.architecturebook.chapter2.testability;

import java.util.Arrays;
import java.util.List;

public class RealDatabaseConnection implements DatabaseConnection {
    public RealDatabaseConnection(String connectionString) {
        System.out.println("\n  [DB] Connecting to... " + connectionString);
    }

    public List<String> getData(String query) {
        return Arrays.asList("real_data_row1", "real_data_row2");
    }
}